src/js/components/product-select/ProductSelect.jsx

```diff
@@ -76,13 +76,12 @@ const ProductSelect = ({
         if (matchedByProductCode) exactMatchProduct = matchedByProductCode;
       }
 
-      if (onExactProductSelected) {
-        onExactProductSelected({ product: exactMatchProduct });
-      }
       if (exactMatchProduct) {
         selectRef.current.select.select.setValue(exactMatchProduct);
       }
-
+      if (onExactProductSelected) {
+        setTimeout(() => onExactProductSelected({ product: exactMatchProduct }), 0);
+      }
```

Comment:
Wrapping `onExactProductSelected` in a **setTimeout** with **0 ms** is kind of a workaround/hack to make this function act asynchronously.
The issue was that on outbound workflow `quantityRequested` field is initially disabled and thus is unfocusable. It turns to enabled only after product is selected so that means this callback would need to be executed after the state change.
Wrapping this callback inside of a **setTimeout** puts it in the **event que** just like the `setState` unlike the regular code that is beeing executed immediately.